ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/LoggingHandler.java

```diff
@@ -34,56 +45,110 @@
-    return skipCodes;
+        .map(Integer::parseInt).collect(ImmutableSet.toImmutableSet());
+  }
+
+  private static Map<String, Double> getSkipRequestPaths(final KsqlRestConfig config) {
+    // Already validated as having a double value
+    return config.getStringAsMap(KSQL_LOGGING_RATE_LIMITED_REQUEST_PATHS_CONFIG)
+        .entrySet().stream()
+        .collect(ImmutableMap.toImmutableMap(Entry::getKey,
+            entry -> Double.parseDouble(entry.getValue())));
   }
 
   @Override
   public void handle(final RoutingContext routingContext) {
-    // If we wanted to log at the beginning of a request, it would go here.
     routingContext.addEndHandler(ar -> {
       // After the response is complete, log results here.
       if (skipResponseCodes.contains(routingContext.response().getStatusCode())) {
         return;
       }
-      String errorMessage = "none";
-      if (routingContext.response().getStatusCode() > 300) {
-        errorMessage = routingContext.response().getStatusMessage();
-        if (Strings.isNullOrEmpty(errorMessage)) {
-          errorMessage = routingContext.getBodyAsString();
+      if (rateLimitedPaths.containsKey(routingContext.request().path())) {
+        final String path = routingContext.request().path();
+        final double rateLimit = rateLimitedPaths.get(path);
+        rateLimiters.computeIfAbsent(path, (k) -> rateLimiterFactory.apply(rateLimit));
+        if (!rateLimiters.get(path).tryAcquire()) {
+          return;
         }
       }
-      logger.accept(String.format(
-          "Request complete - %s %s status: %d, user agent: %s, request body: %d bytes,"
-              + " error response: %s",
+      final long contentLength = routingContext.request().response().bytesWritten();
+      final HttpVersion version = routingContext.request().version();
+      final HttpMethod method = routingContext.request().method();
+      final String uri = routingContext.request().uri();
+      final int status = routingContext.request().response().getStatusCode();
+      final long requestBodyLength = routingContext.request().bytesRead();
+      final String versionFormatted;
+      switch (version) {
+        case HTTP_1_0:
+          versionFormatted = "HTTP/1.0";
+          break;
+        case HTTP_1_1:
+          versionFormatted = "HTTP/1.1";
+          break;
+        case HTTP_2:
+          versionFormatted = "HTTP/2.0";
+          break;
+        default:
+          versionFormatted = "-";
+      }
+      final String name = Optional.ofNullable((ApiUser) routingContext.user())
+          .map(u -> u.getPrincipal().getName())
+          .orElse("-");
+      final String userAgent = Optional.ofNullable(
+          routingContext.request().getHeader(HTTP_HEADER_USER_AGENT)).orElse("-");
+      final String timestamp = Utils.formatRFC1123DateTime(clock.millis());
+      final String message = String.format(
+          "%s - %s [%s] \"%s %s %s\" %d %d \"-\" \"%s\" %d",
```

Comment:
Does it make sense to add a short docs topic on the meaning of this format, since there are adaptations from the standard Apache log format?